File: source/com/barrybecker4/puzzle/common1/ui/PathNavigator.java
// Copyright by Barry G. Becker, 2013. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.common1.ui;


import com.barrybecker4.puzzle.common1.model.Move;

import java.util.List;

/**
 * Allows navigating forward and backward in the solution path.
 * @author Barry Becker
 */
public interface PathNavigator {

    /**
     * @return the path to navigate on.
     */
    List<? extends Move> getPath();

    /**
     * Switch from the current move in the sequence forwards or backwards stepSize.
     * @param currentStep current position in path
     * @param undo whether to make the move or undo it. For some puzzles applying the same move a second time undoes it.
     */
    void makeMove(int currentStep, boolean undo);

}

File: scala-source/com/barrybecker4/puzzle/common/ui/PathNavigator.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.common.ui

import com.barrybecker4.puzzle.common.model.Move


/**
  * Allows navigating forward and backward in the solution path.
  *
  * @author Barry Becker
  */
trait PathNavigator {

  /** @return the path to navigate on. */
  def getPath: List[_ <: Move]

  /**
    * Switch from the current move in the sequence forwards or backwards stepSize.
    *
    * @param currentStep current position in path
    * @param undo        whether to make the move or undo it. For some puzzles applying the same move a second time undoes it.
    */
  def makeMove(currentStep: Int, undo: Boolean): Unit
}
